habitat-lab/habitat/tasks/rearrange/sub_tasks/articulated_object_sensors.py

```diff
@@ -64,6 +71,196 @@ def get_observation(self, observations, episode, *args, **kwargs):
         return np.array(rel_marker_pos)
 
 
+@registry.register_sensor
+class HandleBBoxSensor(UsesArticulatedAgentInterface, Sensor):
+    """
+    Detect handle and return bbox
+    """
+
+    cls_uuid: str = "handle_bbox"
+
+    def __init__(self, sim, config, *args, task, **kwargs):
+        super().__init__(config=config)
+        self._sim = sim
+        self._task = task
+        self._target_height = config.height
+        self._target_width = config.width
+        self._pixel_size = config.pixel_size
+        self._handle_size = config.handle_size
+        self._origin_height = None
+        self._origin_width = None
+
+    @staticmethod
+    def _get_uuid(*args, **kwargs):
+        return HandleBBoxSensor.cls_uuid
+
+    def _get_sensor_type(self, *args, **kwargs):
+        return SensorTypes.TENSOR
+
+    def _get_observation_space(self, *args, config, **kwargs):
+        return spaces.Box(
+            shape=(
+                config.height,
+                config.width,
+                1,
+            ),
+            low=np.finfo(np.float32).min,
+            high=np.finfo(np.float32).max,
+            dtype=np.float32,
+        )
+
+    def _get_image_coordinate_from_world_coordinate(
+        self, point: np.ndarray, target_key: str
+    ) -> Tuple[int, int, bool]:
+        """Project point in the world frame to the image plane"""
+        # Get the camera info
+        fs_w, fs_h, cam_pose = self._get_camera_param(target_key)
+
+        # Do projection from world coordinate into the camera frame
+        point_cam_coord = np.linalg.inv(cam_pose) @ [
+            point[0],
+            point[1],
+            point[2],
+            1,
+        ]
+
+        # From the camera frame into image pixel frame
+        # For image width coordinate
+        w = self._origin_width / 2.0 + (
+            fs_w * point_cam_coord[0] / point_cam_coord[-2]
+        )
+        # For image height coordinate
+        h = self._origin_height / 2.0 + (
+            fs_h * point_cam_coord[1] / point_cam_coord[-2]
```

Comment:
Can we keep indexing consistent?
```suggestion
            fs_w * point_cam_coord[0] / point_cam_coord[2]
        )
        # For image height coordinate
        h = self._origin_height / 2.0 + (
            fs_h * point_cam_coord[1] / point_cam_coord[2]
```